vector/src/main/java/im/vector/app/features/lifecycle/VectorActivityLifecycleCallbacks.kt

```diff
@@ -16,31 +16,127 @@
+        if (activitiesInfo.isEmpty()) {
+            activitiesInfo = packageManager.getPackageInfo(context.packageName, PackageManager.GET_ACTIVITIES).activities
+        }
+
+        // Get all running activities on app task
+        // and compare to activities declared in manifest
+        val manager = context.getSystemService<ActivityManager>() ?: return@withContext false
+
+        return@withContext if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            // Android lint may return an error on topActivity field.
+            // This field was added in ActivityManager.RecentTaskInfo class since Android M (API level 23)
+            // and it is inherited from TaskInfo since Android Q (API level 29).
+            // API 23 changes : https://developer.android.com/sdk/api_diff/23/changes/android.app.ActivityManager.RecentTaskInfo
+            // API 29 changes : https://developer.android.com/sdk/api_diff/29/changes/android.app.ActivityManager.RecentTaskInfo
+            manager.appTasks.any { appTask ->
+                try {
+                    appTask.taskInfo.topActivity?.let { isPotentialMaliciousActivity(context, it) } ?: false
+                } catch (e: IllegalArgumentException) {
+                    // The task was not found. We can ignore it.
+                    Timber.e("The task was not found: ${e.localizedMessage}")
+                    false
+                }
+            }
+        } else {
+            // Android lint may return an error on topActivity field.
+            // This was present in ActivityManager.RunningTaskInfo class since API level 1!
+            // and it is inherited from TaskInfo since Android Q (API level 29).
+            // API 29 changes : https://developer.android.com/sdk/api_diff/29/changes/android.app.ActivityManager.RunningTaskInfo
+            manager.getRunningTasks(10).any { runningTaskInfo -> runningTaskInfo.topActivity?.let { isPotentialMaliciousActivity(context, it) } ?: false }
+        }
     }
 
-    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+    /**
+     * Detect potential malicious activity.
+     * Check if the activity running with package name task affinity is declared in app manifest.
+     *
+     * @param context       the application context
+     * @param activity      the activity of the task
+     * @return true if the activity is potentially malicious
+     */
+    private fun isPotentialMaliciousActivity(context: Context, activity: ComponentName): Boolean {
+        return try {
+            val taskAffinity = context.packageManager.getActivityInfo(activity, 0).taskAffinity
+
+            // Check whether the task affinity matches with the package name
+            if (!taskAffinity.isNullOrEmpty() && taskAffinity == context.packageName) {
+                // Check whether the activity is legitimate (declared in the manifest)
+                !activitiesInfo.any { it.name == activity.className }
```

Comment:
Just a small improvement
```suggestion
                activitiesInfo.none { it.name == activity.className }
```

Reply:
Done